File: Program's_Contributed_By_Contributors/C++/Binary-search.cpp
#include<bits/stdc++.h>
using namespace std;


int binarySearch(int arr[], int left, int right, int element)
{
	if(right>=left)
	{
		int mid = left + (right-left)/2; 

		if(arr[mid] == element) 						
			return mid;

		if(arr[mid] > element)                 				
			return binarySearch(arr,left,mid-1,element);

		return binarySearch(arr,mid+1,right,element);

	}
	return -1;
}

int main()
{
	int size;
	cin>>size;
	int arr[size];
	for(int i=0;i<size;i++){
		cin>>a[i];
	}
	int element = 12;

	int result = binarySearch(arr,0,size-1,element);
	if(result==-1)
		cout<<"Element is not present in array"<<endl;
	else
		cout<<"Element is present @ index "<<result<<endl;

	return 0;
}

File: binary-search.java
<script>
// Iterative function to implement Binary Search
let iterativeFunction = function (arr, x) {

	let start=0, end=arr.length-1;
		
	// Iterate while start not meets end
	while (start<=end){

		// Find the mid index
		let mid=Math.floor((start + end)/2);

		// If element is present at mid, return True
		if (arr[mid]===x) return true;

		// Else look in left or right half accordingly
		else if (arr[mid] < x)
			start = mid + 1;
		else
			end = mid - 1;
	}

	return false;
}

// Driver code
let arr = [1, 3, 5, 7, 8, 9];
let x = 5;

if (iterativeFunction(arr, x, 0, arr.length-1))
	document.write("Element found!<br>");
else document.write("Element not found!<br>");

x = 6;

if (iterativeFunction(arr, x, 0, arr.length-1))
	document.write("Element found!<br>");
else document.write("Element not found!<br>");
</script>									
